binding/Binding/SKManagedStream.cs

```diff
@@ -87,20 +87,22 @@ protected override void DisposeManaged ()
 
 		private IntPtr OnReadManagedStream (IntPtr buffer, IntPtr size)
 		{
-			if (buffer == IntPtr.Zero)
-				throw new ArgumentNullException (nameof (buffer));
 			if ((int)size < 0)
 				throw new ArgumentOutOfRangeException (nameof (size));
 
 			if (size == IntPtr.Zero)
 				return IntPtr.Zero;
 
+			// NOTE: some skips still requires a read as some streams cannot seek
 			using var managedBuffer = Utils.RentArray<byte> ((int)size);
 			var len = stream.Read (managedBuffer.Array, 0, managedBuffer.Length);
```

Comment:
This is not perfect, but I will fix this in a separate PR. The annoying thing is that some streams like and say they are seekable but are not or they allow seeking past the end of the stream causing other issues...